spec/controllers/rooms_controller_spec.rb

```diff
@@ -206,6 +206,14 @@ def random_valid_room_params
       expect(flash[:alert]).to be_present
       expect(response).to redirect_to(root_path)
     end
+
+    it "should not allow the user to join if the user isn't signed in and room authentication is required" do
+      allow_any_instance_of(Setting).to receive(:get_value).and_return("true")
+
+      post :join, params: { room_uid: @room }
+
+      expect(response).to render_template(:join)
```

Comment:
This doesn't actually test the a join used can't join a room. It just test if a join template will render (which it will regardless of what the setting is set to). This can be fixed once you make the change to the redirect